copt/utils.py

```diff
@@ -77,6 +77,45 @@ def safe_sparse_add(a, b):
                 b = b.ravel()
         return a + b
 
+@njit(nogil=True)
+def fast_csr_vm(x, data, indptr, indices, d, idx):
+    """
+
+    :param x:
+    :param data:
+    :param indptr:
+    :param indices:
+    :param d:
+    :param idx:
+    :return:
+    """
+    res = np.zeros(d)
+    assert x.shape[0] == len(idx)
+    for k, i in np.ndenumerate(idx):
+        for j in range(indptr[i], indptr[i+1]):
+            j_idx = indices[j]
+            res[j_idx] += x[k] * data[j]
+    return res
+
+
+@njit(nogil=True)
+def fast_csr_mv(data, indptr, indices, x, idx):
+    """
```

Comment:
please add description of what the function does

Reply:
Done